src/app/screens/Login/index.js

```diff
@@ -23,25 +25,41 @@ class LoginContainer extends Component {
         this.setState({ invalidPassword: password.length < 8 });
     };
 
-    loginOnPressHandler = () => {
-        const { navigation } = this.props;
-        navigation.navigate(Home);
+    loginOnPressHandler = async () => {
+        const { navigation, signIn } = this.props;
+        const { email, password } = this.state;
+        await signIn(email, password);
+        const { signInErrorMessage } = this.props;
+        return !signInErrorMessage && navigation.replace(Home);
     };
 
     render() {
+        const { signInErrorMessage } = this.props;
+        const { email, password, invalidEmail, invalidPassword } = this.state;
+        const disableSubmit = !email.length || !password.length || password.length < 8 || invalidEmail || invalidPassword;
+        
         return (
             <Login 
                 { ...this.state }
                 validateEmail={this.validateEmail} 
                 onChangeEmail={this.onChangeEmail}
                 validatePassword={this.validatePassword}
                 onChangePassword={this.onChangePassword}
-                disableSubmit={!this.state.email.length || !this.state.password.length || this.state.password.length < 8 || this.state.invalidEmail || this.state.invalidPassword}
+                disableSubmit={disableSubmit}
                 onPress={this.loginOnPressHandler}
+                signInErrorMessage={signInErrorMessage}
             />
         );
     }
 }
 
-export default LoginContainer;
+const mapStateToProps = state => ({
+    signInErrorMessage: state.signInErrorMessage
```

Comment:
i think you receive this value from `state.auth.signInErrorMessage` isntead

Reply:
We test both cases and this works ok. I think it's because of how he defined and created the redux store. Maybe he will change this when he add another reducer or he can change it now to avoid future problems

Reply:
Yes, i think it's because the root reducer is a single reducer (auth reducer) and not a combined one. The value `state.auth.signInErrorMessage` return `undefined`.